lms/djangoapps/courseware/courses.py

```diff
@@ -629,78 +629,141 @@ def get_course_assignments(course_key, user, include_access=False):  # lint-amne
+        'required': True
+    }]
+    valid_assessments = block_data.get_xblock_field(ora_block, 'valid_assessments')
+    if valid_assessments:
+        all_assessments.extend(valid_assessments)
+
+    # Loop through all steps and construct Assignment tuples from them
+    assignments = []
+    for assessment in all_assessments:
+        assignment = _ora_assessment_to_assignment(
+            block_data,
+            ora_block,
+            complete,
+            assessment
+        )
+        if assignment is not None:
+            assignments.append(assignment)
     return assignments
 
 
+def _ora_assessment_to_assignment(
+    block_data,
+    ora_block,
+    complete,
+    assessment
+):
+    """
+    Create an assignment from an ORA assessment dict
+    """
+    date_config_type = block_data.get_xblock_field(ora_block, 'date_config_type', 'manual')
+    assignment_type = block_data.get_xblock_field(ora_block, 'format', None)
+    block_title = block_data.get_xblock_field(ora_block, 'title', _('Open Response Assessment'))
+    course_key = block_data.root_block_usage_key
+
+    # Steps with no "due" date, like staff or training, should not show up here
+    assessment_step_due = assessment.get('start')
+    if assessment_step_due is None:
+        return None
+
+    if date_config_type == 'subsection':
+        assessment_start = block_data.get_xblock_field(ora_block, 'start')
+        assessment_due = block_data.get_xblock_field(ora_block, 'due')
+        extra_info = None
+    elif date_config_type == 'course_end':
+        assessment_start = None
+        assessment_due = block_data.get_xblock_field(course_key, 'end')
+        extra_info = None
+    else:
+        assessment_start, assessment_due = None, None
+        if assessment.get('start'):
+            assessment_start = parse_date(assessment.get('start')).replace(tzinfo=pytz.UTC)
+        if assessment.get('due'):
+            assessment_due = parse_date(assessment.get('due')).replace(tzinfo=pytz.UTC)
+        extra_info = _(
+            "This Open Response Assessment's due dates are set by your instructor and can't be shifted."
+        )
+
+    if assessment_due is None:
+        return None
+
+    assessment_name = assessment.get('name')
+    if assessment_name is None:
+        return None
+
+    if assessment_name == 'self-assessment':
+        assessment_type = _("Self Assessment")
+    elif assessment_name == 'peer-assessment':
+        assessment_type = _("Peer Assessment")
+    elif assessment_name == 'staff-assessment':
+        assessment_type = _("Staff Assessment")
+    elif assessment_name == 'submission':
+        assessment_type = _("Submission")
+    else:
+        assessment_type = assessment_name
+    title = f"{block_title} ({assessment_type})"
+    url = ''
+    now = datetime.now(pytz.UTC)
+    assignment_released = not assessment_start or assessment_start < now
+    if assignment_released:
+        url = reverse('jump_to', args=[course_key, ora_block])
```

Comment:
Hey @jansenk, This is Arslan from MIT Open Learning.

We recently noticed an issue where the ORA submission links from the `Dates` tab weren't working. Every ORA link from the dates would end up in a 404 error. So, We just wanted to check if we are missing something or if this might be a general bug.

While looking at the problem closely we noticed that there might be a problem with the generated `jump_to` link in this case. The problem is that there is no course key in the URL or not at the correct place. Here are some samples:

**_Here is a sample link from dates for non-ORA problems:_**

```
http://localhost:18000/courses/course-v1:MIT+TEST_ORA_1+1/jump_to/block-v1:MIT+TEST_ORA_1+1+type@sequential+block@21b0b56c939a482ca80b0f62fc72cbcd
```

**_Here is a sample link from dates for ORA problems:_**



```
http://localhost:18000/courses/block-v1:MIT+TEST_ORA_1+1+type@course+block@course/jump_to/block-v1:MIT+TEST_ORA_1+1+type@openassessment+block@7c58ff39466744afa9646e2ac4f72ecb```

It seems that the problem is with `/courses/<course_id>`. In the case of ORA submissions, it is adding a block key in the URL instead of the course key. Looks like [This line of code](https://github.com/openedx/edx-platform/pull/33118/files#diff-ef3015fcebc131986629db8289d0f77bb5f2b608196bec30373c251a4627c4f8R702) above is problematic since it is obtaining block key instead of course key.

If you can remember, Could you give some insights?